server/backend/api/app/routers/router_guards.py

```diff
@@ -438,3 +441,10 @@ def enforce_bceid_terms_conditions_guard(
             error_code=ERROR_CODE_TERMS_CONDITIONS_REQUIRED,
             error_msg="Requires to accept terms and conditions.",
         )
+
+
+def get_api_instance_env(
+    application_id: int, db: Session = Depends(database.get_db)
+) -> ApiInstanceEnv:
+    application = crud_application.get_application(db, application_id)
+    return crud_utils.use_api_instance_by_app(application)
```

Comment:
Hi Catherine, can this be in `router_utils.py`?

Reply:
um... I was thinking of that before, but then feel it can be a router dependency, so the idim router apis can get the api instance directly
<img width="1160" alt="Screen Shot 2024-07-18 at 3 52 24 PM" src="https://github.com/user-attachments/assets/961b355e-d100-4bde-b710-587d39c0ea22">

Reply:
Oh, sorry, you probably misunderstood.
I meant, putting "get_api_instance_env()" in "router_utils.py" can still be used in the router as dependency.